application-admintools-default/src/main/java/com/xwiki/admintools/internal/data/ConfigurationDataProvider.java

```diff
@@ -108,44 +108,45 @@ private String getJavaVersion()
     }
 
     /**
-     * Identify the used database for XWiki by verifying the configration files.
+     * Identify the used database for XWiki by accessing the {@link DatabaseMetaData}.
      *
-     * @return the name of the used database or {@code null} in case an error occurred or the used DB is not supported.
+     * @return the name and version of the used database or {@code null} in case an error occurred or the used DB is
+     * not supported.
      */
-    private String identifyDB() throws Exception
+    private String identifyDB()
     {
         String usedDB = null;
         try {
-            ServerIdentifier server = getCurrentServer();
-            String databaseCfgPath = server.getXwikiCfgFolderPath() + "hibernate.cfg.xml";
-            String patternString = "<property name=\"connection.url\">jdbc:(.*?)://";
-            Pattern pattern = Pattern.compile(patternString);
-            File file = new File(databaseCfgPath);
-            BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
-
-            while (bufferedReader.ready()) {
-                String line = bufferedReader.readLine();
-                Matcher matcher = pattern.matcher(line);
-                if (matcher.find()) {
-                    String foundDB = matcher.group(1);
-                    usedDB = this.currentServer.getSupportedDBs().getOrDefault(foundDB, null);
-                    break;
-                }
+            DatabaseMetaData metaData = getDBMetaData();
+            if (metaData != null) {
+                usedDB = metaData.getDatabaseProductName() + " - " + metaData.getDatabaseProductVersion();
```

Comment:
Here as well. Instead of the concatenated data, it's better to keep a map of values that it's concatenated at display. You can also think that this is a API and someone else might use this values in a different why than you, as you are using the Ping class, and they will need to do a split on these values
So, instead of `usedDB`, you will have a `DBMetadata` (or something similar)